skada/_subspace.py

```diff
@@ -366,3 +375,277 @@ def TransferComponentAnalysis(
+        """Predict adaptation (weights, sample or labels).
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        X_t : array-like, shape (n_samples, n_components)
+            The data transformed to the target subspace.
+        y_t : array-like, shape (n_samples,)
+            The labels (same as y).
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels transformed to the target subspace
+            (same as sample_domain).
+        weights : None
+            No weights are returned here.
+        """
+        X, sample_domain = check_X_domain(
+            X,
+            sample_domain,
+            allow_multi_source=True,
+            allow_multi_target=True,
+        )
+        X_source, X_target = source_target_split(X, sample_domain=sample_domain)
+
+        if np.array_equal(X_source, self.X_source_) and np.array_equal(
+            X_target, self.X_target_
+        ):
+            K = self._get_kernel_matrix(X_source, X_target)
+            X_ = K @ self.A_
+        else:
+            Ks = pairwise_kernels(X, self.X_source_, metric=self.kernel)
+            Kt = pairwise_kernels(X, self.X_target_, metric=self.kernel)
+            K = np.concatenate((Ks, Kt), axis=1)
+            X_ = K @ self.A_
+        return X_
+
+    def _get_mmd_matrix(self, ns, nt, sample_domain):
+        Mss = (1 / (ns**2)) * np.ones((ns, ns))
+        Mtt = (1 / (nt**2)) * np.ones((nt, nt))
+        Mst = -(1 / (ns * nt)) * np.ones((ns, nt))
+        M = np.block([[Mss, Mst], [Mst.T, Mtt]])
+        return M
+
+    def _get_kernel_matrix(self, X_source, X_target):
+        Kss = pairwise_kernels(X_source, metric=self.kernel)
+        Ktt = pairwise_kernels(X_target, metric=self.kernel)
+        Kst = pairwise_kernels(X_source, X_target, metric=self.kernel)
+        K = np.block([[Kss, Kst], [Kst.T, Ktt]])
+        return K
+
+    def fit(self, X, y=None, sample_domain=None, **kwargs):
+        """Fit adaptation parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, sample_domain = check_X_domain(
+            X,
+            sample_domain,
+            allow_multi_source=True,
+            allow_multi_target=True,
+        )
+        X_source, X_target = source_target_split(X, sample_domain=sample_domain)
+
+        if self.n_components is None:
```

Comment:
Can you change the choice of `n_component` for the SA method? It is not the same as TCA. Except it was wanted @kachayev